src/utils/c_utils/src/string_c_utils.c

```diff
@@ -464,10 +464,13 @@ int ADUC_AllocAndStrCopyN(char** dest, const char* src, size_t srcLen)
         goto done;
     }
 
+    // Uses ADUC_Safe_StrCopyN to prevent buffer overflows and ensure null termination.
```

Comment:
not optimal, but would it be worthwhile to be defensive and have this method do a strlen(src) and fail if srcLen > strlen(src)?

Reply:
This is a great suggestion. Pushed a commit that implements this along with obligatory unit tests. Also, added 4-byte utf-8 test case for duck emoji.